### PR TITLE
CAMEL-16275: move RouteBuilderLoader implementations to dedicated artefact

### PR DESCRIPTION
**Pull request by lburgazzoli**

### CODE REVIEW

**Comment by lburgazzoli** [comment]
@davsclaus I had to move camel-main tests related to XML from the camel-main module to XMLs modules to avoid cyclic dependencies between core and dsl

**Review by oscerd — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
Yeah I think that is okay.

The tests for camel-main is then focusing more on just the functionality of camel-main itself. And then the dsl loader is in the new dsl modules.

Also moving it here is actually great as we have all the catalog information so we can better build the DSL.

**Comment by lburgazzoli** [comment]
I'm adding the same kind of tests for the other DSLs

File: dsl/camel-xml-jaxb-dsl/pom.xml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    <parent>
+        <groupId>org.apache.camel</groupId>
+        <artifactId>dsl</artifactId>
+        <version>3.9.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>camel-xml-jaxb-dsl</artifactId>
+    <packaging>jar</packaging>
+    <name>Camel :: XML DSL with camel-xml-jaxb</name>
+    <description>Camel XML DSL with acmel-xml-jaxb</description>
```
Comment (davsclaus) [inline dsl/camel-xml-jaxb-dsl/pom.xml:34]: Typo

File: bom/camel-bom/pom.xml [hunk 2]
```diff
@@ -947,6 +947,11 @@
         <artifactId>camel-jasypt</artifactId>
         <version>${project.version}</version>
       </dependency>
+			<dependency>
```
Comment (davsclaus) [inline bom/camel-bom/pom.xml:950]: To many spaces here

**Review by davsclaus — APPROVED** [review]

**Comment by davsclaus** [comment]
LGTM

**Pull request merged**